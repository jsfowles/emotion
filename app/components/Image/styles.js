import glamorous from 'glamorous';

import * as c from '../../identity/constants';

export const Img = glamorous.img(
  {
    maxWidth: '100%',
    opacity: 0,
    transition: '600ms opacity ease',
  },
  ({ loaded }) => loaded && ({ opacity: 1 }),
  ({ fill }) => fill && ({ width: '100%' }),
  ({ absolute }) => absolute && ({
    ...c.ABSOLUTE,
    objectFit: 'cover',
  }),
);
