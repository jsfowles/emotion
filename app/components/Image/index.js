import React from 'react';
import PropTypes from 'prop-types';

import { Img } from './styles';
import ImageLoader from '../../containers/ImageLoader';

const Image = ({
  src,
  absolute,
  alt,
  fill,
  onLoad,
  style,
}) => (
  <ImageLoader src={ src }>
    { source => (
      <Img
        onLoad={ onLoad }
        src={ source }
        alt={ alt }
        loaded={ source }
        absolute={ absolute }
        fill={ fill }
        style={ style }
      />
    ) }
  </ImageLoader>
);

Image.defaultProps = {
  aspectRatio: 55.5,
  framed: false,
  absolute: false,
  alt: null,
  onLoad: null,
  fill: false,
  style: null,
};

Image.propTypes = {
  src: PropTypes.string.isRequired,
  aspectRatio: PropTypes.number,
  framed: PropTypes.bool,
  absolute: PropTypes.bool,
  fill: PropTypes.bool,
  alt: PropTypes.string,
  onLoad: PropTypes.func,
  style: PropTypes.object, // eslint-disable-line
};

export default Image;
