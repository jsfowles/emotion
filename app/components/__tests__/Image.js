import React from 'react';
import renderer from 'react-test-renderer';

import Image from '../Image';

describe('Image', () => {
  test('renders', () => {
    const tree = renderer.create(<Image src="http://via.placeholder.com/350x150" />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
