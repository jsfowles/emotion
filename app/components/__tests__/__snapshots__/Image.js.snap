// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image renders 1`] = `
<img
  alt={null}
  className="css-1p3alxs"
  onLoad={null}
  src={null}
  style={null}
/>
`;
